xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-webjar/src/main/webjar/searchSuggest.js

```diff
@@ -106,19 +106,22 @@ var XWiki = (function (XWiki) {
      * Submits the form or go to a selected page according to selection.
      */
     onSuggestionSelected: function(event) {
-      if (event.memo.suggest == this.suggest) {
+      if (event.memo.suggest === this.suggest) {
         event.stop();
-        // Also stop the browser event that triggered the custom "xwiki:suggest:selected" event.
-        if (event.memo.originalEvent) {
-          Event.stop(event.memo.originalEvent);
-        }
         if (!event.memo.url) {
+          // Stop the browser event that triggered the custom "xwiki:suggest:selected" event.
+          if (event.memo.originalEvent.type === 'click' && event.memo.originalEvent) {
+            Event.stop(event.memo.originalEvent);
+          }
           // Submit form
           this.searchInput.up('form').submit();
         }
         else {
-          // Go to page
-          window.location = event.memo.url;
+          // The page anchor should be able to handle its own click event (event.memo.originalEvent here) itself.
+          if (event.memo.originalEvent.type !== 'click' && event.memo.originalEvent) {
```

Comment:
```suggestion
          if (event.memo.originalEvent && event.memo.originalEvent.type !== 'click') {
```

Same as above, I don't see why you would check if the original event exists after accessing a property of it. This could also be an excellent use case of [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).